Resources/input/fsh/profiles/0_GEM_PR_KIM_AdresseIdentifier.fsh

```diff
@@ -0,0 +1,49 @@
+Profile: GEM_PR_KIM_AdressIdentifier
+Parent: Identifier
+Id: GEM-PR-MEDREQ-KIM-Adresse
+Title: "GEM PR MEDREQ KIM-Adresse"
+Description: "Identifier zur Angabe der KIM-Adresse"
+* insert Meta (GEM-PR-MEDREQ-KIM-Adresse)
+* system = $kim (exactly)
+* system 1..1 MS
+* value 1..1
```

Comment:
@simoneOnFhir Sollten wir ein Constraint hinzufügen, der in Form einer Regex überprüft, ob es sich um eine valide KIM Addresse handelt? Wie würde diese Regex aussehen?

Reply:
Gute Idee! Wie lautet das Schema für gültige KIM-Adressen?

Reply:
Regeln wie für mail Adressen. Am Ende des domain part haben wir aktuell immer kim.telematik. Das kann sich mit KIM 2.0 aber ändern.
-> Also constraint wie für Mails. Hast du was zum abschreiben?

Reply:
https://stackoverflow.com/questions/201323/how-can-i-validate-an-email-address-using-a-regular-expression

leider bei weitem komplexer als erhofft :-/ (und wahrscheinlich ziemlich hässlich, für SUSHI zu escapen)
Allerdings berücksichtigt das die komplette eMail-Spezifikation, einschließlich IP-Adressen als Domäne etc. 
Eine RegEx speziell für KIM könnte vermutlich erheblich einfacher sein...

Reply:
Ich kriege den Constraint nicht platziert. Hier meine Versuche:

**1. Variante**
```
* value 1..1
     * ^regex = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"
```
-> Sushi Error: The element or path you referenced does not exist: regex

**2. Variante**
```
* value 1..1
     * ^pattern = "(?i)[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,}"
```
-> Sushi Error: The element or path you referenced does not exist: pattern 

**3. Variante**
```
* value 1..1
  * ^binding.valueSet = "http://hl7.org/fhir/ValueSet/contact-point-system|email"
```
->  HAPI Error: 
"StructureDefinition.differential.element[1].binding | ElementDefinition.binding.strength: mindestens erforderlich = 1, aber nur gefunden 0"

Reply:
Ja, nee, das wäre ja auch zu einfach :D

1. Schritt: Invariante mit FHIR-Path-Expression bauen:
```
Invariant: KIM-adresse-validieren
Description: "Dies ist keine gültge KIM-Adresse"
Expression: "matches('^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$')"
Severity: #warning
```
Severity kann auch auf #error gesetzt werden, wenn die Validierung nucht nur eine Warnung erzeugen, sondern komplett fehlschlagen soll.

2. Schritt: Invariante an der gewünschen Stelle (ggf auch an verschiedenen Stellen) einbinden
``` 
* value 1..1
* value obeys KIM-adresse-validieren
```

Reply:
Hinweis: Was ihr unter Description angebt, wird bei der Validierung als Fehlermeldung ausgeworfen, wenn eine Instanz den Constraint verletzt.